From 82f9db2fd8b8c0a321376af8ca1016bec91633a7 Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@ciq.com>
Date: Tue, 23 Jun 2026 11:33:01 -0700
Subject: [PATCH] Revert "mm/gup: reintroduce pin_user_pages_fast_only()"

This reverts commit ef467f3c43338f72f4173c98b15bc8021f79c3a1.

This helper is no longer used by the FIPS-mode RNG, which was the
motivation for reintroducing it. Remove it.

Signed-off-by: Sultan Alsawaf <sultan@ciq.com>
---
 include/linux/mm.h |  2 --
 mm/gup.c           | 14 --------------
 2 files changed, 16 deletions(-)

diff --git a/include/linux/mm.h b/include/linux/mm.h
index 3f1772eb2426..ab100f6bd25a 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -2571,8 +2571,6 @@ int get_user_pages_fast(unsigned long start, int nr_pages,
 			unsigned int gup_flags, struct page **pages);
 int pin_user_pages_fast(unsigned long start, int nr_pages,
 			unsigned int gup_flags, struct page **pages);
-int pin_user_pages_fast_only(unsigned long start, int nr_pages,
-			     unsigned int gup_flags, struct page **pages);
 void folio_add_pin(struct folio *folio);
 
 int account_locked_vm(struct mm_struct *mm, unsigned long pages, bool inc);
diff --git a/mm/gup.c b/mm/gup.c
index 40b8a69fc312..16cdddef9158 100644
--- a/mm/gup.c
+++ b/mm/gup.c
@@ -3359,20 +3359,6 @@ int pin_user_pages_fast(unsigned long start, int nr_pages,
 }
 EXPORT_SYMBOL_GPL(pin_user_pages_fast);
 
-/*
- * This is the FOLL_PIN equivalent of get_user_pages_fast_only(). Behavior is
- * the same, except that this one sets FOLL_PIN instead of FOLL_GET.
- */
-int pin_user_pages_fast_only(unsigned long start, int nr_pages,
-			     unsigned int gup_flags, struct page **pages)
-{
-	if (!is_valid_gup_args(pages, NULL, &gup_flags,
-			       FOLL_PIN | FOLL_FAST_ONLY))
-		return -EINVAL;
-	return internal_get_user_pages_fast(start, nr_pages, gup_flags, pages);
-}
-EXPORT_SYMBOL_GPL(pin_user_pages_fast_only);
-
 /**
  * pin_user_pages_remote() - pin pages of a remote process
  *
-- 
2.52.0

